packages/core/card.ts

```diff
@@ -508,147 +561,69 @@ export class Card {
 
   private async applyRulesForAttribute(
     attributes: AttributesObject,
-    field: string,
-    rules: OcclusionRulesOrDefaults,
+    field: FieldCard,
+    rules: OcclusionRules,
     data: any,
-    included: Map<string, OcclusionRulesOrDefaults[]>
+    included: Map<string, OcclusionRules[]>
   ): Promise<void> {
-    let includeFields = await this.includeFieldsForField(rules, field);
-    if (includeFields === 'everything' || includeFields === 'upstream') {
-      attributes[field] = cloneDeep(data);
-    } else if (rules) {
-      if (!includeFields.length) {
-        return;
-      }
-      if (includeFields.find(i => i.name === field) && !isObjectLike(data)) {
-        attributes[field] = data;
-        return;
-      }
+    if (await field.hasFeature('field-deserialize')) {
+      // this field's value is not a Card, it's a scalar. So no deeper occlusion
+      // is needed, the whole value is in.
+      attributes[field.name] = data;
+      return;
+    }
 
-      let interiorIncludeFields = flatten(
-        typeof includeFields !== 'string' ? includeFields.map(i => i.includeFields || []) : []
-      );
-      let interiorRule: OcclusionRulesOrDefaults =
-        typeof includeFields !== 'string' ? { includeFields: interiorIncludeFields } : includeFields;
-      setInteriorCardFieldSet(rules, interiorRule);
-
-      if (Array.isArray(data)) {
-        attributes[field] = await Promise.all(
-          data.map(async cardValue => {
-            let interiorCard = await this.container.instantiate(
```

Comment:
Instantiation here is not always right. It's better to use `this.value()`, which will respect the custom deserializer.